### PR TITLE
Nextion - Remove soft reset at start

### PR DESCRIPTION
**Pull request by edwardtfn**
# What does this implement/fix?

This introduces a user configurable attribute to Nextion display which allows skipping the soft reset on displays during the component startup.
I don't know really why this reset was introduced from the beginning, so I've left as a configurable attribute to avoid a breaking change, but on my tests it always work without this command and it makes the startup process a bit faster (my main goal).

This is not a breaking change, as the previous behavior is the default.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** solves esphome/feature-requests#2074

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#3566

## Test Environment

- [X] ESP32
- [X] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
display:
  - id: disp1
    platform: nextion
    soft_reset_on_start: false
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by kbx81 — APPROVED** [review]
LGTM -- thanks!

File: esphome/components/nextion/display.py [hunk 1]
```diff
@@ -72,6 +73,7 @@
             cv.Optional(CONF_START_UP_PAGE): cv.positive_int,
             cv.Optional(CONF_AUTO_WAKE_ON_TOUCH, default=True): cv.boolean,
             cv.Optional(CONF_EXIT_REPARSE_ON_START, default=False): cv.boolean,
+            cv.Optional(CONF_SOFT_RESET_ON_START, default=True): cv.boolean,
```
Comment (kbx81) [inline esphome/components/nextion/display.py:76]: Hmm, I don't see where this is set in `to_code`

**Review by kbx81 — CHANGES_REQUESTED** [review]
Just noticed something...

File: esphome/components/nextion/display.py (hunk 1 above)
Comment (edwardtfn) [reply esphome/components/nextion/display.py:76]: I'm not sure I got your question.

Comment (kbx81) [reply esphome/components/nextion/display.py:76]: In the yaml, if I define `soft_reset_on_start: false`, how/where does this value get passed to/set in the underlying C code?

Comment (edwardtfn) [reply esphome/components/nextion/display.py:76]: Now I understand why I haven't noticed any difference with this. 😆 

I've just fixed with https://github.com/esphome/esphome/pull/6134/commits/3557439f68c550d998213abd76f158ce22e132d9, but now I will have to run tests again before resubmit this PR.
Thanks for your catch!

**Comment by edwardtfn** [comment]
I'm dropping this PR as the display requires this `rest` to announce itself to the system. It does work without it, but then it requires a manual set to "Ready" status, but then the queue gets unsync.
Maybe I come back to this in the future.

**Pull request closed without merging**